CONTRIBUTING.md

```diff
@@ -0,0 +1,239 @@
+```sh
+$ git clone https://github.com/<your username>/modmail
+...
+$ cd modmail
+```
+
+or using the [github cli](https://github.com/cli/cli):
+
+```sh
+$ gh repo clone <your username>/modmail
+...
+$ cd modmail
+```
+
+> Tip: You can use the github cli to fork the repo as well, just use `gh repo fork discord-modmail/modmail` and it will allow you to clone it directly.
+
+### Install development dependencies
+
+Make sure you are in the project directory.
+
+```sh
+# This will install the development and project dependencies.
+poetry install
+# This will install the pre-commit hooks.
+poetry run task precommit
+```
+
+### Set up environment variables
+
+1. Create a text file named `.env` in your project root (that's the base folder of your repository):
+   - You can also copy the `.env.template` file to `.env`
+
+> Note: The entire file name is literally `.env`
+
+1. Open the file with any text editor.
+1. Each environment variable is on its own line, with the variable and the value separated by a `=` sign.
+
+#### The following variables are needed for running Modmail:
+
+| Required | ENV VARIABLE NAME | TYPE    | WHAT IS IT?                                      |
+| -------- | ----------------- | ------- | ------------------------------------------------ |
+| True     | `TOKEN`           | String  | Bot Token from the Discord developer portal      |
+| False    | `DEVELOP`         | Boolean | Enables the developer bot extensions             |
+| False    | `PLUGIN_DEV`      | Boolean | Enables plugin-developer friendly bot extensions |
+
+The rest of them can be viewed in our [.env.template](./.env.template)
+
+### Run The Project
+
+To run the project, use the (below) in the project root.
+
+```shell
+$ poetry run task start
+```
+
+## Contribute Code
+
+We like code commits a lot! They're super handy, and they keep the project going and doing the work it needs to do to be useful to others.
+
+Code contributions of just about any size are acceptable!
+
+To contribute code:
+
+- [Set up the project](#project-setup).
+- Make any necessary changes to the source code.
+- Write clear, concise commit message(s).
+  - A more in-depth guide to writing great commit messages can be found in Chris Beam's [*How to Write a Git Commit Message*](https://chris.beams.io/posts/git-commit/).
+- Run `flake8`, `black` and `pre-commit` against your code **before** you push. Your commit will be rejected by the build server if it fails to lint. You can run the lint by executing `poetry run task lint` in your command line.
+- Go to [discord-modmail/modmail/pulls][modmail-pulls] and open a new pull request with your changes.
+- If PRing from your own fork, **ensure that "Allow edits from maintainers" is checked**. This permits maintainers to commit changes directly to your fork, speeding up the review process.
+- If your PR is connected to an open issue, add a line in your PR's description that says `Closes #123`, where `#123` is the number of the issue you're fixing.
```

Comment:
... (This will make github link your issue and make it easier for us and other contributers to find and understand your PR )